toolkits/postgres/arcade_postgres/tools/postgres.py

```diff
@@ -0,0 +1,155 @@
+from typing import Annotated, Any
+
+from arcade_tdk import ToolContext, tool
+from arcade_tdk.errors import RetryableToolError
+from sqlalchemy import inspect, text
+from sqlalchemy.ext.asyncio import AsyncEngine
+
+from .database_engine import ERROR_REMAPPING, MAX_ROWS_RETURNED, DatabaseEngine
+
+
+@tool(requires_secrets=["DATABASE_CONNECTION_STRING"])
+async def discover_tables(
+    context: ToolContext,
+    schema_name: Annotated[str, "The database schema to discover tables in"] = "public",
+) -> list[str]:
+    """Discover all the tables in the postgres database when the list of tables is not known.
+
+    THIS TOOL SHOULD ALWAYS BE USED BEFORE ANY OTHER TOOL THAT REQUIRES A TABLE NAME.
+    """
+    async with await DatabaseEngine.get_engine(
+        context.get_secret("DATABASE_CONNECTION_STRING")
+    ) as engine:
+        tables = await _get_tables(engine, schema_name)
+        return tables
+
+
+@tool(requires_secrets=["DATABASE_CONNECTION_STRING"])
+async def get_table_schema(
+    context: ToolContext,
+    schema_name: Annotated[str, "The database schema to get the table schema of"],
```

Comment:
possible future feature: I wonder if a discover_schemas/list_schemas tool would be useful? As someone who did not create the db, I'm not sure which schema names are available to me.